honeybee_ies/cli/translate.py

```diff
@@ -25,16 +26,26 @@ def translate():
     type=click.Path(exists=False, file_okay=False, resolve_path=True,
                     dir_okay=True), default='.', show_default=True
 )
-def model_to_gem(model_json, name, folder):
+@click.option(
+    '--honeybee/--dragonfly', is_flag=True, help='A flag to indicate if the input '
+    'file is a honeybee or a dragonfly file. Default is a Honeybee file.',
+    default=True
+)
```

Comment:
Thanks! I thought about this but it won't work with the pickled files. We want a solution not to load the whole file first.

Reply:
I didn't know that you also wanted this to work with the .pkl files but you can definitely get this to work using the same solution we use in honeybee-core and dragonfly-core to sense the file type:

```
import json
import pickle

# sense the file type from the first character to avoid maxing memory with JSON
with open(model_file) as inf:
    first_char = inf.read(1)
is_json = True if first_char == '{' else False
if is_json:
with open(model_json) as inf:
    data = json.load(inf) if is_json else pickle.load(inf)

# sense the model type from the keys available in the dictionary
if 'buildings' in data or 'context_shades' in data:
    model = DFModel.from_dict(data)
else:  # it's a HBJSON
    model = HBModel.from_dict(data)
```